File: sentimental-cash/cash.py
from cs50 import get_float # for learning to import smth
from typing import Final # for "constants" better usage are enums


def main():
    print(calculate_change(ask_for_change()))


def ask_for_change():
    while True:
        change = input("Give me my change: $")
        try:
            change = float(change)  # for learning to handle invalid inputs
        except ValueError:
            print("Usage x.xx numerical")
        else:
            if change >= 0.00:
                print("0k, heres your change:", end="")
                break
            else:
                print("no negative change! :)")
    return change


def calculate_change(change):
    # dictionarys are like arrays
    COIN: Final = {"QUARTER":  25, "DIME": 10, "NICKEL": 5, "PENNY": 1}
    coins = {"quarters": 0, "dimes": 0, "nickels": 0, "pennys": 0}

    change = change*100 # for rounding fails

    while change >= COIN["QUARTER"]:
        coins["quarters"] += 1
        change -= COIN["QUARTER"]
    while change >= COIN["DIME"]:
        coins["dimes"] += 1
        change -= COIN["DIME"]  # like c algorh
    while change >= COIN["NICKEL"]:
        coins["nickels"] += 1
        change -= COIN["NICKEL"]
    while change >= COIN["PENNY"]:
        coins["pennys"] += 1
        change -= COIN["PENNY"]

    return sum(coins.values())  # keys = "..."  :X
 #gives back all values ^            values =    ^


main()

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_count(int change);
int give_me_my_change(void);

int main(void)
{
    printf("%i\n", calculate_count(give_me_my_change()));
}

int give_me_my_change(void)
{
    int change;
    // user input
    do
    {
        change = get_int("Give me my Change! My Change is: ");
    }
    while (change < 0); // rejecting negatives
    return change;
}

int calculate_count(int change)
{
    // contants for readability
    const int quarter = 25;
    const int dime = 10;
    const int nickel = 5;
    const int cent = 1;

    int quarters = 0, dimes = 0, nickels = 0, cents = 0; // counters

    // calculate quarters
    while (change >= quarter)
    {
        quarters++;
        change -= quarter;
    }
    // calculate dimes
    while (change >= dime)
    {
        dimes++;
        change -= dime;
    }
    // calculate nickels
    while (change >= nickel)
    {
        nickels++;
        change -= nickel;
    }
    // calculate cents
    while (change >= cent)
    {
        cents++;
        change -= cent;
    }

    int sum = quarters + dimes + nickels + cents;
    return sum;
}
